### PR TITLE
Auto pack template in build.cmd

### PR DESCRIPTION
**Pull request by qinezh**
@chenkennt @superyyrrzz @vwxyzh @hellosnow @ansyral

### CODE REVIEW

File: UpdateTemplate.cmd [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+SETLOCAL
+SETLOCAL ENABLEDELAYEDEXPANSION
+
+:: Check if node exists globally
+WHERE node >NUL
+IF NOT '%ERRORLEVEL%'=='0' (
+    ECHO ERROR: UpdateTemplate.cmd requires node installed globally.
+    GOTO :Exit
+)
+
```
Comment (superyyrrzz) [inline UpdateTemplate.cmd]: Also check gulp?

Comment (qinezh) [inline UpdateTemplate.cmd]: No need to check gulp since the step `npm install` will install it and other dependency

Comment (superyyrrzz) [inline UpdateTemplate.cmd]: But I meet the error `'gulp' is not recognized as an internal or external command,
operable program or batch file.`... Seems need to install gulp globally?

Comment (qinezh) [inline UpdateTemplate.cmd]: You're right, I should add a step to global install gulp

File: UpdateTemplate.cmd [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+PUSHD %DefaultTemplate%
+CALL npm install
+CALL gulp
+POPD
+
+PUSHD %TemplateHome%
+CALL npm install
+CALL gulp
+POPD
```
Comment (chenkennt) [inline UpdateTemplate.cmd]: missed one popd in the end

Comment (chenkennt) [inline UpdateTemplate.cmd]: actually only one pair of pushd and popd is needed at beginning and end of the script.

---
In reply to: [66747755](https://github.com/dotnet/docfx/pull/379#discussion_r66747755) [](ancestors = 66747755)

File: UpdateTemplate.cmd [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+PUSHD %~dp0
+
+SETLOCAL
+SETLOCAL ENABLEDELAYEDEXPANSION
+
+:: Check if node exists globally
+WHERE node >NUL
+IF NOT '%ERRORLEVEL%'=='0' (
+    ECHO ERROR: UpdateTemplate.cmd requires node installed globally.
+    GOTO :Exit
```
Comment (hellosnow) [inline UpdateTemplate.cmd]: add definition for Exit?

**Comment by hellosnow** [comment]
:shipit:

**Comment by superyyrrzz** [comment]
:+1:

**Comment by superyyrrzz** [comment]
@DuncanmaMSFT We've removed the ZIPs from our repo, as `build.cmd` will generate them automatically. So the ZIPs are no longer required in commits.

**Pull request merged**